--- 7save/tmgl7b.t.00	2021-08-22 19:52:06.430761392 -0400
+++ 7save/tmgl7b.t.01	2021-08-22 19:46:27.148767835 -0400
@@ -30,7 +30,7 @@
 /* - finishing the output with a newline */
 /* (c) 2020, Andrii Makukha, 2-clause BSD License. */
 
-begin:	ignore(blanks) [wordsz = &classes - &nclass]
+begin:	ignore(blanks)
 pr1:	comment\pr1
 	parse(first)\pr2
 	diag(error)
@@ -43,14 +43,13 @@
 first:	parse(( fref = { <start: rc x > 1 *}))
 	getfref line = { 2 <:> 1 };
 
-error:	smark ignore(none) any(!<<>>) string(!<<;>>) scopy 
+error:	smark ignore(none) any(all) string(nosemi) scopy 
 	( <;> = {<;>} | null )
 	= { * <??? error: > 2 1 * };
 
 line:
 	labels
-	( charcl <;>
-	| statement
+	( statement
 	| numbers
 	| trule <;> )
 	= { 2 * 1 * };
@@ -65,7 +64,7 @@
 last:	= { <st=0100> * <fi=0200> * };
 
 comment: </*>
-co1:	ignore(!<<*>>) <*> ignore(none) </>/co1;
+co1:	ignore(nostar) <*> ignore(none) </>/co1;
 
 statement: [csym=0] oldtab(dtt) oldtab(pat)
 	( prc plst tlst <)> = (1){2 1 }
@@ -120,13 +119,12 @@
 
 pdot:	<.>/done ignore(none) ident\alias
 	([dtt?] | table(dtt) [ndt=0]) [ndt++];
-spdot:	<.> ignore(none) not(( any(letter) ))
+spdot:	<.> ignore(none) not(( char(letter) ))
 alias:	newtab(dtt,ndt);
 
 parg:	rname | remote(specparg);
 
 specparg: number
-	| charcl
 	| <<> longlit
 	| <*> = { <\n> }
 	| <(> ( <)> = { <x no;> }
@@ -249,46 +247,6 @@
 	| <~> = {<cm>}
 	);
 
-charcl:
-	( <!> ccla cclb | ccla )
-	octal(classmask);
-ccla:	(<<<>) [classmask = 1<<nclass++] [classmask?]/cherr
-ccl1:	cclc <<<>\ccl1;
-cclc:	ignore(none)
-
-ccl3:	<>>\ccl4 ccle\ccl3;
-ccl4:	(<>> | cclx fail) (not((<>>)) | ccle);
-ccle:	char(n) [*(wordsz*n+&classes) =| classmask];
-cclb:	zeron
-ccl5:	[*(wordsz*n+&classes) =^ classmask] testn\ccl5;
-cclx:	[nclass--] zeron
-ccl6:	[*(wordsz*n+&classes) =& ~classmask] testn\ccl6;
-cherr:	diag(( ={<too many char classes>} ));
-
-zeron:	[n=0];
-testn:	[++n<400?];
-
-classmask: 0;
-nclass:	0;
-classes:
-cl1:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl2:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl3:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl4:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl5:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl6:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl7:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl8:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-/* More classes added to avoid issues with UTF-8 or Unicode */
-cl9:	0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl10:   0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl11:   0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl12:   0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl13:   0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl14:   0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl15:   0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-cl16:   0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;
-
 done:	;
 
 create:	[csym++]
@@ -305,11 +263,11 @@
 remote:	params(1) create parse(rem1,$1);
 rem1:	params(1) getcsym $1 = { 2 <=.> * 1 * };
 
-number: smark ignore(none) any(digit) string(digit) scopy;
+number: smark ignore(none) char(digit) string(digit) scopy;
 
 name:	ident scopy;
 
-ident:	smark ignore(none) any(letter) string(alpha);
+ident:	smark ignore(none) char(letter) string(alpha);
 
 oldtab:	params(1) [$1?]/done discard($1) [$1=0];
 
@@ -327,16 +285,16 @@
 
 nil:	{};
 
-blanks:	<< 	
-	>>;
-digit:	<<0123456789>>;
-letter:	<<abcdefghijklmnopqrstuvwxyz>>
-	<<ABCDEFGHIJKLMNOPQRSTUVWXYZ>>;
-alpha:	<<0123456789>>
-	<<abcdefghijklmnopqrstuvwxyz>>
-	<<ABCDEFGHIJKLMNOPQRSTUVWXYZ>>;
-litch:	!<<\>>>;
-none:	<<>>;
+/* character classes */
+blanks: 000600; 000000; 400000; 000000; 000000; 000000; 000000; 000000;
+digit:	000000; 000000; 000000; 777400; 000000; 000000; 000000; 000000;
+letter:	000000; 000000; 000000; 000000; 377774; 777600; 377774; 777600;
+alpha:	000000; 000000; 000000; 777400; 377774; 777600; 377774; 777600;
+litch:	777777; 777777; 777777; 777767; 777777; 777777; 777777; 777777;
+nostar:	777777; 777777; 777577; 777777; 777777; 777777; 777777; 777777;
+none:	000000; 000000; 000000; 000000; 000000; 000000; 000000; 000000;
+all:	777777; 777777; 777777; 777777; 777777; 777777; 777777; 777777;
+nosemi:	777777; 777777; 777777; 777677; 777777; 777777; 777777; 777777;
 
 csym:	0;
 fsym:	0;
@@ -350,5 +308,3 @@
 npa:	0;	/*number of parsing rule params*/
 ptt:	0;	/*table of params of translation*/
 npt:	0;	/*number of params of translation*/
-i:	0;
-wordsz: 0;      /* word size on the current architecture */
