--- tmgl7a.t	2021-08-22 22:03:01.507309529 -0400
+++ tmgl7b.t.save	2021-08-22 20:21:43.530310865 -0400
@@ -43,7 +43,7 @@
 first:	parse(( fref = { <start: rc x > 1 *}))
 	getfref line = { 2 <:> 1 };
 
-error:	smark ignore(none) any(!<<>>) string(!<<;>>) scopy
+error:	smark ignore(none) any(all) string(nosemi) scopy 
 	( <;> = {<;>} | null )
 	= { * <??? error: > 2 1 * };
 
@@ -64,7 +64,7 @@
 last:	= { <st=0100> * <fi=0200> * };
 
 comment: </*>
-co1:	ignore(!<<*>>) <*> ignore(none) </>/co1;
+co1:	ignore(nostar) <*> ignore(none) </>/co1;
 
 statement: [csym=0] oldtab(dtt) oldtab(pat)
 	( prc plst tlst <)> = (1){2 1 }
@@ -81,7 +81,7 @@
 
 pident:	ident newtab(pat,npa);
 
-tlst:	<;>/null [i=0] list((name [i =+ 1])) remote((octal(i)))
+tlst:	<;>/null [i=0] list((name [i=+ 1])) remote((octal(i)))
 	= { <push;> 1 * 2 * };
 
 frag:	prule = (1){ 1(nil,q1) }
@@ -119,7 +119,7 @@
 
 pdot:	<.>/done ignore(none) ident\alias
 	([dtt?] | table(dtt) [ndt=0]) [ndt =+ 1];
-spdot:	<.> ignore(none) not(( any(letter) ))
+spdot:	<.> ignore(none) not(( char(letter) ))
 alias:	newtab(dtt,ndt);
 
 parg:	rname | remote(specparg);
@@ -131,7 +131,7 @@
 		| push(3,dtt,ndt,sndt) [dtt=0]
 			prule <)> oldtab(dtt)
 			( ifelem = {1(nil,xbit) }
-			| = {1(nil,nil)*<x no;>}
+			| = {1(nil,nil)*<x no;>} 
 		)	);
 
 iseasy:	[easy = 1];
@@ -200,11 +200,10 @@
 		| rva = { 2 * 1 }
 		| () );
 rva:	<?> rv <:> rv fref fref 
-	= { <_t;alt;> * 2 * 4 * <salt;> * 1 * 2 <:> 3 * 1 <:> };
+	= { <_t;alt;> * 2 * 4 * <salt;> 1 * 2 <:> 3 * 1 <:> };
 
 prime:
-	lv suffix/done = { 2 * 1 }
-	| prefix lv = { 1 * 2 }
+	<&> lv = { <rm 1; ro fi; addr> }
 	| <(> expr <)> 
 	| unary prime = { 1 * 2 }
 	| remote(number) = { <_l;> 1 };
@@ -233,14 +232,6 @@
 		| = {<gt>} )
 	);
 
-prefix:	smark ignore(none)
-	( <&> = {<_lvXXX>}
-	);
-
-suffix:	smark ignore(none)
-	( <--> = {<_daXXX>}
-	);
-
 unary:	( <-> = {<mi>}
 	| <~> = {<cm>}
 	);
@@ -248,7 +239,7 @@
 done:	;
 
 create:	[csym =+ 1]
-getcsym: octal(csym) = { <.> 1 };
+getcsym: octal(csym) = {<.> 1};
 
 fref:	[fsym =+ 1]
 getfref: octal(fsym) = { <..> 1 };
@@ -261,16 +252,16 @@
 remote:	params(1) create parse(rem1,$1);
 rem1:	params(1) getcsym $1 = { 2 <=.> * 1 * };
 
-number: smark ignore(none) any(digit) string(digit) scopy;
+number: smark ignore(none) char(digit) string(digit) scopy;
 
 name:	ident scopy;
 
-ident:	smark ignore(none) any(letter) string(alpha);
+ident:	smark ignore(none) char(letter) string(alpha);
 
 oldtab:	params(1) [$1?]/done discard($1) [$1=0];
 
 newtab:	params(2) ([$2?] | table($2) [$1=0])
-	enter($2,i) [$2[i] = $1] [$1 =+ 1];
+	enter($2,i) [$2[i] = $1 =+ 1];
 
 tabval:	params(2) [$2?] find($2,i) [i=$1-$2[i]] octal(i);
 
@@ -283,16 +274,16 @@
 
 nil:	{};
 
-blanks:	<< 	
-	>>;
-digit:	<<0123456789>>;
-letter:	<<abcdefghijklmnopqrstuvwxyz>>
-	<<ABCDEFGHIJKLMNOPQRSTUVWXYZ>>;
-alpha:	<<0123456789>>
-	<<abcdefghijklmnopqrstuvwxyz>>
-	<<ABCDEFGHIJKLMNOPQRSTUVWXYZ>>;
-litch:	!<<\>>>;
-none:	<<>>;
+/* character classes */
+blanks: 000600; 000000; 400000; 000000; 000000; 000000; 000000; 000000;
+digit:	000000; 000000; 000000; 777400; 000000; 000000; 000000; 000000;
+letter:	000000; 000000; 000000; 000000; 377774; 777600; 377774; 777600;
+alpha:	000000; 000000; 000000; 777400; 377774; 777600; 377774; 777600;
+litch:	777777; 777777; 777777; 777767; 777777; 777777; 777777; 777777;
+nostar:	777777; 777777; 777577; 777777; 777777; 777777; 777777; 777777;
+none:	000000; 000000; 000000; 000000; 000000; 000000; 000000; 000000;
+all:	777777; 777777; 777777; 777777; 777777; 777777; 777777; 777777;
+nosemi:	777777; 777777; 777777; 777677; 777777; 777777; 777777; 777777;
 
 csym:	0;
 fsym:	0;
@@ -306,4 +297,3 @@
 npa:	0;	/*number of parsing rule params*/
 ptt:	0;	/*table of params of translation*/
 npt:	0;	/*number of params of translation*/
-i:	0;
